packages/web-components/fast-foundation/src/custom-properties/behavior.ts

```diff
@@ -18,8 +18,120 @@ export interface CSSCustomPropertyTarget {
     disconnectedCSSCustomPropertyRegistry: CSSCustomPropertyDefinition[] | void;
 }
 
-export type CSSCustomPropertyBehavior = Behavior & CSSCustomPropertyDefinition;
+/**
+ * A Behavior that, when bound, will register itself with the nearest
+ * CSSCustomPropertyTarget
+ */
+export class CSSCustomPropertyBehavior implements Behavior, CSSCustomPropertyDefinition {
+    /**
+     * The name of the CSS Custom Property behavior.
+     */
+    public readonly name: CSSCustomPropertyDefinition["name"];
+
+    /**
+     * The value or function that will resolve the value of
+     * the CSS Custom Property.
+     */
+    public readonly value: CSSCustomPropertyDefinition["value"];
+    constructor(
+        /**
+         * The name of the custom property.
+         * @remarks
+         * When written as a custom property, the "--" required
+         * by CSS custom properties will be prepended to this value.
+         */
+        name: string,
+
+        /**
+         * The value of the custom property or a function that
+         * resolves the value.
+         * @remarks
+         * When a resolver is provided, the resolver will be invoked
+         * by the element returned from the host argument.
+         */
+        value: CSSCustomPropertyDefinition["value"],
+
+        /**
+         * The element that registers the behavior.
+         * @remarks
+         * This element should also be responsible for resolving
+         * and function value.
+         */
+        host: (
+            source: typeof FASTElement & HTMLElement
+        ) => Partial<CSSCustomPropertyTarget> | null
+    ) {
+        this.name = name;
+        this.value = value;
+        this.host = host;
+    }
+
+    private host: (
+        source: typeof FASTElement & HTMLElement
+    ) => Partial<CSSCustomPropertyTarget> | null;
+
+    private _propertyName: string;
+    /**
+     * The name of the CSSCustomPropertyBehavior formatted
+     * as a CSS custom property.
+     */
+    public get propertyName() {
```

Comment:
Any reason not to assign `propertyName `and `var` in the constructor when the name is set?

Reply:
Probably not a good one. The idea was to lazily evaluate it but the operation is so cheap that its probably better to just do the work in the constructor, reduce the amount of coded needed to do the work and remove the runtime check.

I'll update.